sql/expression/function/json_set_test.go

```diff
@@ -73,18 +73,21 @@ func TestJSONSet(t *testing.T) {
 		{f1, sql.Row{json, "$.c.d", "test"}, `{"a": 1, "b": [2, 3], "c": {"d": "test"}}`, nil},                   // update existing nested
 		{f2, sql.Row{json, "$.a", 10, "$.e", "new"}, `{"a": 10, "b": [2, 3], "c": {"d": "foo"},"e":"new"}`, nil}, // update existing and set new
 		{f1, sql.Row{json, "$.a.e", "test"}, `{"a": 1, "b": [2, 3], "c": {"d": "foo"}}`, nil},                    // set new nested does nothing
-		{f1, sql.Row{json, "$.b[0]", 4}, `{"a": 1, "b": [4, 3], "c": {"d": "foo"}}`, nil},                        // update element in array
-		// {f1, sql.Row{json, "$.b[5]", 4}, `{"a": 1, "b": [2, 3, 4], "c": {"d": "foo"}}`, nil},                        // update element in array out of range
-		// {f1, sql.Row{json, "$.b.c", 4}, `{"a": 1, "b": [2, 3], "c": {"d": "foo"}}`, nil},                        // set nested in array does nothing
-		// {f1, sql.Row{json, "$.a[0]", 4}, `{"a": 4, "b": [2, 3], "c": {"d": "foo"}}`, nil}, // update single element with indexing
-		// {f1, sql.Row{json, "$.a[5]", 4}, `{"a": [1, 4], "b": [2, 3], "c": {"d": "foo"}}`, nil},   // update single element with indexing out of range
-		// {f1, sql.Row{json, "$[0]", 4}, `4`, nil},   // struct indexing
+		{f1, sql.Row{json, "$.c.e", "test"}, `{"a": 1, "b": [2, 3], "c": {"d": "foo","e":"test"}}`, nil},         // set new nested in existing struct
+		// {f1, sql.Row{json, "$.c[5]", 4}, `{"a": [1, 4], "b": [2, 3], "c": {"d": "foo"}}`, nil},                   // update struct with indexing out of range
+		{f1, sql.Row{json, "$.b[0]", 4}, `{"a": 1, "b": [4, 3], "c": {"d": "foo"}}`, nil},   // update element in array
+		{f1, sql.Row{json, "$.b[5]", 4}, `{"a": 1, "b": [2, 3,4], "c": {"d": "foo"}}`, nil}, // update element in array out of range
+		{f1, sql.Row{json, "$.b.c", 4}, `{"a": 1, "b": [2, 3], "c": {"d": "foo"}}`, nil},    // set nested in array does nothing
+		{f1, sql.Row{json, "$.a[0]", 4}, `{"a": 4, "b": [2, 3], "c": {"d": "foo"}}`, nil},   // update single element with indexing
+		// {f1, sql.Row{json, "$.a[5]", 4}, `{"a": [1, 4] , "b": [2, 3], "c": {"d": "foo"}}`, nil}, // update single element with indexing out of range
```

Comment:
Do these commented out cases fail today? If we want to put pause on this work, I think it'd be preferable for unsupported cases to throw an error instead of returning an incorrect result. I'm open to continuing to get the rest of these working if you think we can wrap them up quickly, otherwise I think it's fine to press pause after this iteration if we think we've gotten the most important cases and the other cases throw good "unsupported" errors or something similar.